### ISSUE TITLE
Never ending "Configuring GHC..."

### ISSUE THREAD

**Issue by gleb-dianov** [commercialhaskell/stack#3712]
### General summary/comments

I have been using stack for a while on this system and everything worked fine, but recently something broke. 

### Steps to reproduce

I'm running the latest version of stack on Arch Linux (Linux 4.14.8-1-ARCH). The problem occurs when I try to execute any commands that require ghc (for example `stack build` or `stack ghci`).

### Expected

The project to build/ghci to open.

### Actual

```
$ stack build --verbose
Version 1.6.3, Git revision b27e629b8c4ce369e3b8273f04db193b060000db (5454 commits) x86_64 hpack-0.20.0
2017-12-27 00:30:13.554735: [debug] Checking for project config at: /media/storage/drive/coding/projects/haider/stack.yaml
@(Stack/Config.hs:842:9)
2017-12-27 00:30:13.554939: [debug] Loading project config file stack.yaml
@(Stack/Config.hs:868:13)
2017-12-27 00:30:13.555907: [debug] Decoding build plan from: /home/gleb/.stack/build-plan/lts-9.5.yaml
@(Stack/Snapshot.hs:150:5)
2017-12-27 00:30:13.555970: [debug] Trying to decode /home/gleb/.stack/build-plan-cache/lts-9.5.cache
@(Data/Store/VersionTagged.hs:66:5)
2017-12-27 00:30:13.561408: [debug] Success decoding /home/gleb/.stack/build-plan-cache/lts-9.5.cache
@(Data/Store/VersionTagged.hs:70:13)
2017-12-27 00:30:13.561779: [debug] Run process: /sbin/ldconfig -p
@(System/Process/Log.hs:37:3)
2017-12-27 00:30:13.582767: [debug] Process finished in 20ms: /sbin/ldconfig -p
@(System/Process/Log.hs:44:3)
2017-12-27 00:30:13.583050: [debug] Run process: /usr/bin/gcc -v
@(System/Process/Log.hs:37:3)
2017-12-27 00:30:13.584277: [debug] Process finished in 1ms: /usr/bin/gcc -v
@(System/Process/Log.hs:44:3)
2017-12-27 00:30:13.584347: [debug] PIE disabled
@(Stack/Setup.hs:589:17)
2017-12-27 00:30:13.589657: [debug] Found shared library libtinfo.so.5 in /usr/lib
@(Stack/Setup.hs:578:38)
2017-12-27 00:30:13.589777: [debug] Found shared library libtinfo.so.6 in 'ldconfig -p' output
@(Stack/Setup.hs:565:29)
2017-12-27 00:30:13.589878: [debug] Found shared library libncursesw.so.6 in 'ldconfig -p' output
@(Stack/Setup.hs:565:29)
2017-12-27 00:30:13.589993: [debug] Found shared library libgmp.so.10 in 'ldconfig -p' output
@(Stack/Setup.hs:565:29)
2017-12-27 00:30:13.590158: [debug] Found shared library libgmp.so.3 in /usr/lib
@(Stack/Setup.hs:578:38)
2017-12-27 00:30:13.590244: [debug] Using tinfo6-nopie GHC build
@(Stack/Setup.hs:620:9)
2017-12-27 00:30:13.848443: [info] Preparing to install GHC (tinfo6-nopie) to an isolated location.
@(Stack/Setup.hs:877:5)
2017-12-27 00:30:13.848549: [info] This will not interfere with any system-level installation.
@(Stack/Setup.hs:886:5)
2017-12-27 00:30:13.849010: [debug] Downloading from https://github.com/commercialhaskell/ghc/releases/download/ghc-8.0.2-release/ghc-8.0.2-x86_64-fedora24-linux.tar.xz to /home/gleb/.stack/programs/x86_64-linux/ghc-tinfo6-nopie-8.0.2.tar.xz ...
@(Stack/Setup.hs:1460:5)
2017-12-27 00:30:13.849089: [debug] Will check against sha1 hash: 926b9ae1448df7894c0c4d5791d5eb5161a9cb52
@(Stack/Setup.hs:1474:13)
Already downloaded.                             
2017-12-27 00:30:14.079240: [debug] menv = fromList [("APPROOT","https://localhost:3443"),("BROWSER","yandex-browser-beta"),("COLORFGBG","15;0"),("COLORTERM","truecolor"),("DBUS_SESSION_BUS_ADDRESS","unix:path=/run/user/1000/bus"),("DISPLAY",":0"),("EDITOR","vim"),("HOME","/home/gleb"),("INVOCATION_ID","c4de387d09c1476c8ea2bf4c764fde02"),("JOURNAL_STREAM","9:19500"),("KONSOLE_DBUS_SERVICE",":1.6"),("KONSOLE_DBUS_SESSION","/Sessions/1"),("KONSOLE_DBUS_WINDOW","/Windows/1"),("KONSOLE_PROFILE_NAME","Profile 2"),("LANG","en_US.UTF-8"),("LANGUAGE",""),("LOGNAME","gleb"),("MAIL","/var/spool/mail/gleb"),("MOZ_PLUGIN_PATH","/usr/lib/mozilla/plugins"),("OLDPWD","/home/gleb"),("PATH","/usr/lib/hardening-wrapper/bin:/usr/lib/hardening-wrapper/bin:/usr/local/sbin:/usr/local/bin:/usr/bin:/usr/lib/jvm/default/bin:/usr/bin/site_perl:/usr/bin/vendor_perl:/usr/bin/core_perl:/usr/local/sbin:/usr/local/bin:/usr/bin:/usr/lib/jvm/default/bin:/usr/bin/site_perl:/usr/bin/vendor_perl:/usr/bin/core_perl:/home/gleb/.gem/ruby/2.3.0/bin:/home/gleb/.vimpkg/bin:/home/gleb/.vimpkg/bin:/home/gleb/.gem/ruby/2.3.0/bin:/home/gleb/perl5/perlbrew/etc/bashrc:/home/gleb/.stack/programs/x86_64-linux/ghc-7.10.2/bin:/home/gleb/.local/bin:/home/gleb/.yarn-global/bin:/home/gleb/.yarn-global/bin:/usr/lib/jvm/default/bin:/usr/bin/site_perl:/usr/bin/vendor_perl:/usr/bin/core_perl:/usr/local/sbin:/usr/local/bin:/usr/bin:/usr/lib/jvm/default/bin:/usr/bin/site_perl:/usr/bin/vendor_perl:/usr/bin/core_perl:/home/gleb/.gem/ruby/2.3.0/bin:/home/gleb/.vimpkg/bin:/home/gleb/.vimpkg/bin:/home/gleb/.gem/ruby/2.3.0/bin:/home/gleb/perl5/perlbrew/etc/bashrc:/home/gleb/.stack/programs/x86_64-linux/ghc-7.10.2/bin:/home/gleb/.local/bin:/home/gleb/.yarn-global/bin:/home/gleb/.yarn-global/bin"),("PROFILEHOME",""),("PWD","/media/storage/drive/coding/projects/haider"),("SHELL","/bin/bash"),("SHELL_SESSION_ID","d3cb50b644324da9bf1faec60d351f51"),("SHLVL","3"),("TERM","xterm-256color"),("TERMINAL","konsole"),("USER","gleb"),("WINDOWID","18874374"),("WINDOWPATH","1"),("XAUTHORITY","/home/gleb/.Xauthority"),("XDG_RUNTIME_DIR","/run/user/1000"),("XDG_SEAT","seat0"),("XDG_SESSION_ID","c1"),("XDG_VTNR","1"),("_","/home/gleb/.local/bin/stack"),("drive","/media/storage/drive")]
@(Stack/Setup.hs:1010:5)
2017-12-27 00:30:14.079765: [debug] ziptool: xz
@(Stack/Setup.hs:1028:5)
2017-12-27 00:30:14.079815: [debug] make: make
@(Stack/Setup.hs:1029:5)
2017-12-27 00:30:14.079862: [debug] tar: tar
@(Stack/Setup.hs:1030:5)
2017-12-27 00:30:14.079933: [debug] Unpacking /home/gleb/.stack/programs/x86_64-linux/ghc-tinfo6-nopie-8.0.2.tar.xz
@(Stack/Setup.hs:1057:5)
2017-12-27 00:30:14.080093: [debug] Run process: /usr/bin/tar Jxf /home/gleb/.stack/programs/x86_64-linux/ghc-tinfo6-nopie-8.0.2.tar.xz
@(System/Process/Log.hs:37:3)
2017-12-27 00:30:23.868854: [debug] Process finished in 9788ms: /usr/bin/tar Jxf /home/gleb/.stack/programs/x86_64-linux/ghc-tinfo6-nopie-8.0.2.tar.xz
@(System/Process/Log.hs:44:3)
2017-12-27 00:30:23.869257: [debug] Run process: /home/gleb/.stack/programs/x86_64-linux/ghc-tinfo6-nopie-8.0.2.temp/ghc-8.0.2/configure --prefix=/home/gleb/.stack/programs/x86_64-linux/ghc-tinfo6-nopie-8.0.2/
@(System/Process/Log.hs:37:3)
Configuring GHC ...
```

```
$ stack ghci --verbose
Version 1.6.3, Git revision b27e629b8c4ce369e3b8273f04db193b060000db (5454 commits) x86_64 hpack-0.20.0
2017-12-27 00:49:45.240240: [debug] Checking for project config at: /home/gleb/stack.yaml
@(Stack/Config.hs:842:9)
2017-12-27 00:49:45.240417: [debug] Checking for project config at: /home/stack.yaml
@(Stack/Config.hs:842:9)
2017-12-27 00:49:45.240473: [debug] Checking for project config at: /stack.yaml
@(Stack/Config.hs:842:9)
2017-12-27 00:49:45.240527: [debug] No project config file found, using defaults.
@(Stack/Config.hs:872:13)
2017-12-27 00:49:45.241184: [debug] Run from outside a project, using implicit global project config
@(Stack/Config.hs:559:13)
2017-12-27 00:49:45.241387: [debug] Using resolver: lts-10.1 from implicit global project's config file: /home/gleb/.stack/global-project/stack.yaml
@(Stack/Config.hs:573:32)
2017-12-27 00:49:45.241453: [debug] Decoding build plan from: /home/gleb/.stack/build-plan/lts-10.1.yaml
@(Stack/Snapshot.hs:150:5)
2017-12-27 00:49:45.241498: [debug] Trying to decode /home/gleb/.stack/build-plan-cache/lts-10.1.cache
@(Data/Store/VersionTagged.hs:66:5)
2017-12-27 00:49:45.246227: [debug] Success decoding /home/gleb/.stack/build-plan-cache/lts-10.1.cache
@(Data/Store/VersionTagged.hs:70:13)
2017-12-27 00:49:45.250000: [debug] Run process: /sbin/ldconfig -p
@(System/Process/Log.hs:37:3)
2017-12-27 00:49:45.251831: [debug] Process finished in 1ms: /sbin/ldconfig -p
@(System/Process/Log.hs:44:3)
2017-12-27 00:49:45.252041: [debug] Run process: /usr/bin/gcc -v
@(System/Process/Log.hs:37:3)
2017-12-27 00:49:45.253264: [debug] Process finished in 1ms: /usr/bin/gcc -v
@(System/Process/Log.hs:44:3)
2017-12-27 00:49:45.253352: [debug] PIE disabled
@(Stack/Setup.hs:589:17)
2017-12-27 00:49:45.257694: [debug] Found shared library libtinfo.so.5 in /usr/lib
@(Stack/Setup.hs:578:38)
2017-12-27 00:49:45.257797: [debug] Found shared library libtinfo.so.6 in 'ldconfig -p' output
@(Stack/Setup.hs:565:29)
2017-12-27 00:49:45.257888: [debug] Found shared library libncursesw.so.6 in 'ldconfig -p' output
@(Stack/Setup.hs:565:29)
2017-12-27 00:49:45.257964: [debug] Found shared library libgmp.so.10 in 'ldconfig -p' output
@(Stack/Setup.hs:565:29)
2017-12-27 00:49:45.258097: [debug] Found shared library libgmp.so.3 in /usr/lib
@(Stack/Setup.hs:578:38)
2017-12-27 00:49:45.258139: [debug] Using tinfo6-nopie GHC build
@(Stack/Setup.hs:620:9)
2017-12-27 00:49:45.485243: [info] Preparing to install GHC (tinfo6-nopie) to an isolated location.
@(Stack/Setup.hs:877:5)
2017-12-27 00:49:45.485348: [info] This will not interfere with any system-level installation.
@(Stack/Setup.hs:886:5)
2017-12-27 00:49:45.485800: [debug] Downloading from https://github.com/commercialhaskell/ghc/releases/download/ghc-8.2.2-release/ghc-8.2.2-x86_64-fedora27-linux.tar.xz to /home/gleb/.stack/programs/x86_64-linux/ghc-tinfo6-nopie-8.2.2.tar.xz ...
@(Stack/Setup.hs:1460:5)
2017-12-27 00:49:45.485854: [debug] Will check against sha1 hash: 955325a4d551b0da3b06d8569dbfc417b8a01af3
@(Stack/Setup.hs:1474:13)
Already downloaded.                             
2017-12-27 00:49:46.029133: [debug] menv = fromList [("APPROOT","https://localhost:3443"),("BROWSER","yandex-browser-beta"),("COLORFGBG","15;0"),("COLORTERM","truecolor"),("DBUS_SESSION_BUS_ADDRESS","unix:path=/run/user/1000/bus"),("DISPLAY",":0"),("EDITOR","vim"),("HOME","/home/gleb"),("INVOCATION_ID","c4de387d09c1476c8ea2bf4c764fde02"),("JOURNAL_STREAM","9:19500"),("KONSOLE_DBUS_SERVICE",":1.6"),("KONSOLE_DBUS_SESSION","/Sessions/1"),("KONSOLE_DBUS_WINDOW","/Windows/1"),("KONSOLE_PROFILE_NAME","Profile 2"),("LANG","en_US.UTF-8"),("LANGUAGE",""),("LOGNAME","gleb"),("MAIL","/var/spool/mail/gleb"),("MOZ_PLUGIN_PATH","/usr/lib/mozilla/plugins"),("OLDPWD","/media/storage/drive/coding/projects/haider"),("PATH","/usr/lib/hardening-wrapper/bin:/usr/lib/hardening-wrapper/bin:/usr/local/sbin:/usr/local/bin:/usr/bin:/usr/lib/jvm/default/bin:/usr/bin/site_perl:/usr/bin/vendor_perl:/usr/bin/core_perl:/usr/local/sbin:/usr/local/bin:/usr/bin:/usr/lib/jvm/default/bin:/usr/bin/site_perl:/usr/bin/vendor_perl:/usr/bin/core_perl:/home/gleb/.gem/ruby/2.3.0/bin:/home/gleb/.vimpkg/bin:/home/gleb/.vimpkg/bin:/home/gleb/.gem/ruby/2.3.0/bin:/home/gleb/perl5/perlbrew/etc/bashrc:/home/gleb/.stack/programs/x86_64-linux/ghc-7.10.2/bin:/home/gleb/.local/bin:/home/gleb/.yarn-global/bin:/home/gleb/.yarn-global/bin:/usr/lib/jvm/default/bin:/usr/bin/site_perl:/usr/bin/vendor_perl:/usr/bin/core_perl:/usr/local/sbin:/usr/local/bin:/usr/bin:/usr/lib/jvm/default/bin:/usr/bin/site_perl:/usr/bin/vendor_perl:/usr/bin/core_perl:/home/gleb/.gem/ruby/2.3.0/bin:/home/gleb/.vimpkg/bin:/home/gleb/.vimpkg/bin:/home/gleb/.gem/ruby/2.3.0/bin:/home/gleb/perl5/perlbrew/etc/bashrc:/home/gleb/.stack/programs/x86_64-linux/ghc-7.10.2/bin:/home/gleb/.local/bin:/home/gleb/.yarn-global/bin:/home/gleb/.yarn-global/bin"),("PROFILEHOME",""),("PWD","/home/gleb"),("SHELL","/bin/bash"),("SHELL_SESSION_ID","d3cb50b644324da9bf1faec60d351f51"),("SHLVL","3"),("TERM","xterm-256color"),("TERMINAL","konsole"),("USER","gleb"),("WINDOWID","18874374"),("WINDOWPATH","1"),("XAUTHORITY","/home/gleb/.Xauthority"),("XDG_RUNTIME_DIR","/run/user/1000"),("XDG_SEAT","seat0"),("XDG_SESSION_ID","c1"),("XDG_VTNR","1"),("_","/home/gleb/.local/bin/stack"),("drive","/media/storage/drive")]
@(Stack/Setup.hs:1010:5)
2017-12-27 00:49:46.029608: [debug] ziptool: xz
@(Stack/Setup.hs:1028:5)
2017-12-27 00:49:46.029655: [debug] make: make
@(Stack/Setup.hs:1029:5)
2017-12-27 00:49:46.029702: [debug] tar: tar
@(Stack/Setup.hs:1030:5)
2017-12-27 00:49:46.029828: [debug] Unpacking /home/gleb/.stack/programs/x86_64-linux/ghc-tinfo6-nopie-8.2.2.tar.xz
@(Stack/Setup.hs:1057:5)
2017-12-27 00:49:46.029973: [debug] Run process: /usr/bin/tar Jxf /home/gleb/.stack/programs/x86_64-linux/ghc-tinfo6-nopie-8.2.2.tar.xz
@(System/Process/Log.hs:37:3)
2017-12-27 00:49:57.316705: [debug] Process finished in 11286ms: /usr/bin/tar Jxf /home/gleb/.stack/programs/x86_64-linux/ghc-tinfo6-nopie-8.2.2.tar.xz
@(System/Process/Log.hs:44:3)
2017-12-27 00:49:57.316993: [debug] Run process: /home/gleb/.stack/programs/x86_64-linux/ghc-tinfo6-nopie-8.2.2.temp/ghc-8.2.2/configure --prefix=/home/gleb/.stack/programs/x86_64-linux/ghc-tinfo6-nopie-8.2.2/
@(System/Process/Log.hs:37:3)
Configuring GHC ...
```

No matter how long I wait nothing happens after that, it just says "Configuring GHC...". I tried reinstalling stack, reinstalling libraries, installing older versions of ghc and stack.

Also, stack works fine when I'm logged in as root.

And global ghc installation works fine.

### Stack version

```
$ stack --version
Version 1.6.3, Git revision b27e629b8c4ce369e3b8273f04db193b060000db (5454 commits) x86_64 hpack-0.20.0
```

### Method of installation

Arch repositories.
```
$ sudo pacman -S stack
```

**Comment by mgsloan** [comment]
There have been quite a few problems recently on arch linux.  Search the issue tracker for info and workarounds.  I haven't seen this particular symptom, though, so maybe it's something else.

@borsboom recently put up this branch https://github.com/commercialhaskell/stack/compare/setup-multi-bindists - not sure if it's ready yet since there's no PR.  Possibly consider building stack with this and see if it fixes the problem?

If a prior ghc worked, you should be able to put it on your PATH and use --system-ghc.

**Comment by borsboom** [comment]
This is not an issue I've come across before, so no idea whether that branch or the setup-info YAML mentioned in https://github.com/commercialhaskell/stack/issues/3518#issuecomment-353899981 will help, but could be worth a try.  It's too bad `stack setup --verbose` doesn't also cause the GHC configure step to show all output.

**Comment by mgsloan** [comment]
Good point!  I've opened https://github.com/commercialhaskell/stack/issues/3716 to track implementing that.

**Comment by borsboom** [comment]
@gleb-dianov One thing you might try is adding this to your `~/.stack/config.yaml`:
```
ghc-build: tinfo6
```
If that doesn't help, see what happens if you run the GHC bindist's `configure` manually.  Unpack `/home/gleb/.stack/programs/x86_64-linux/ghc-tinfo6-nopie-8.0.2.tar.xz` and then run `./configure`.

**Comment by gleb-dianov** [comment]
@borsboom I tried changing the config file and it didn't help. I tried running `configure` manually and it looks like the problem is not with stack but with gcc.

```
$ 
[config.log](https://github.com/commercialhaskell/stack/files/1591423/config.log)
./configure 
checking for path to top of build tree... /home/gleb/.stack/programs/x86_64-linux/ghc-8.0.2
Build platform inferred as: x86_64-unknown-linux
Host platform inferred as: x86_64-unknown-linux
Target platform inferred as: x86_64-unknown-linux
GHC build  : x86_64-unknown-linux
GHC host   : x86_64-unknown-linux
GHC target : x86_64-unknown-linux
checking for perl... /usr/bin/perl
checking for a BSD-compatible install... /usr/bin/install -c
checking whether ln -s works... yes
checking for gsed... sed
checking for gcc... /usr/bin/gcc
checking for llc-3.7... no
checking for llc... no
checking for opt-3.7... no
checking for opt... no
checking for ld... /usr/lib/hardening-wrapper/bin/ld
checking for gcc... /usr/bin/gcc
checking whether the C compiler works... no
configure: error: in `/home/gleb/.stack/programs/x86_64-linux/ghc-8.0.2':
configure: error: C compiler cannot create executables
See `config.log' for more details
```

**Comment by borsboom** [comment]
Did `./configure` hang in this case?

**Comment by gleb-dianov** [comment]
@borsboom no, it terminated after it showed that output

**Comment by gleb-dianov** [comment]
Ok, I partly fixed it. I tried installing all the updates on my laptop and checking if it works, and it did. I saw on some forums that people recommend checking the PATH variable, so I did that. I found out that I have a mess there. I copied my `.bashrc` and `.bash_profile` files from my laptop and now it almost works.

I have another problem now, that I was trying to solve in the first place when I got "Configuring GHC..." issue. Here it is:

```
$ stack ghci --verbose 
Version 1.4.0, Git revision e714f1dd3fade19496d91bd6a017e435a96a6bcd (4640 commits) x86_64 hpack-0.17.0
2018-01-07 00:11:35.915509: [debug] Checking for project config at: /home/gleb/stack.yaml
@(Stack/Config.hs:935:9)
2018-01-07 00:11:35.915670: [debug] Checking for project config at: /home/stack.yaml
@(Stack/Config.hs:935:9)
2018-01-07 00:11:35.915723: [debug] Checking for project config at: /stack.yaml
@(Stack/Config.hs:935:9)
2018-01-07 00:11:35.915830: [debug] No project config file found, using defaults.
@(Stack/Config.hs:964:13)
2018-01-07 00:11:35.916801: [debug] Run from outside a project, using implicit global project config
@(Stack/Config.hs:526:13)
2018-01-07 00:11:35.917216: [debug] Using resolver: lts-10.2 from implicit global project's config file: /home/gleb/.stack/global-project/stack.yaml
@(Stack/Config.hs:540:32)
2018-01-07 00:11:35.917298: [debug] Trying to decode /home/gleb/.stack/build-plan-cache/x86_64-linux/lts-10.2.cache
@(Data/Store/VersionTagged.hs:68:5)
2018-01-07 00:11:35.917393: [debug] Exception ignored when attempting to load /home/gleb/.stack/build-plan-cache/x86_64-linux/lts-10.2.cache: /home/gleb/.stack/build-plan-cache/x86_64-linux/lts-10.2.cache: openBinaryFile: does not exist (No such file or directory)
@(Data/Store/VersionTagged.hs:86:9)
2018-01-07 00:11:35.917464: [debug] Failure decoding /home/gleb/.stack/build-plan-cache/x86_64-linux/lts-10.2.cache
@(Data/Store/VersionTagged.hs:75:13)
2018-01-07 00:11:35.917596: [debug] Decoding build plan from: /home/gleb/.stack/build-plan/lts-10.2.yaml
@(Stack/BuildPlan.hs:493:5)
2018-01-07 00:11:37.570894: [debug] Decoding build plan from file failed: AesonException "Error in $.packages.cassava.constraints.flags['bytestring--lt-0_10_4']: Invalid flag name: \"bytestring--lt-0_10_4\""
@(Stack/BuildPlan.hs:498:13)
2018-01-07 00:11:37.571304: [debug] Downloading build plan from: https://raw.githubusercontent.com/fpco/lts-haskell/master//lts-10.2.yaml
@(Stack/BuildPlan.hs:503:13)
2018-01-07 00:11:37.571359: [debug] Downloading /fpco/lts-haskell/master//lts-10.2.yaml
@(Network/HTTP/Download.hs:78:5)
Downloaded lts-10.2 build plan.    
AesonException "Error in $.packages.cassava.constraints.flags['bytestring--lt-0_10_4']: Invalid flag name: \"bytestring--lt-0_10_4\""
```

GHC and GHCi work fine inside projects (with different resolvers), but globally they don't

**Comment by borsboom** [comment]
Looks like in this process you downgraded to stack-1.4.0.  The Cassava flags issue is fixed in 1.6.3.
